### ISSUE TITLE
Migrating to version 6.0.0 broke nested navigation.

### ISSUE THREAD

**Issue by Sevastyan** [Milad-Akarie/auto_route_library#1411]
My nested navigation broke with an error:

```
ConfiguredAppRoute does not have children
'package:auto_route/src/route/route_config.dart':
Failed assertion: line 269 pos 12: 'this[key]?.children != null'
```

My AutoRouter:

```dart
@AutoRouterConfig(
  replaceInRouteName: 'Screen,Route',
)
class MainRouter extends _$MainRouter {
  @override
  RouteType get defaultRouteType => const RouteType.material();

  @override
  final List<AutoRoute> routes = [
    AutoRoute(page: SplashRoute.page, path: '/'),
    AutoRoute(page: NoCredentialsRoute.page),
    AutoRoute(page: ProvisioningRoute.page),
    AutoRoute(
      page: ConfiguredAppRoute.page,
      children: [
        AutoRoute(page: SessionRoute.page, path: ''),
        AutoRoute(page: UpdatingRoute.page),
      ],
    ),
  ];
}
```

Place of use:

```dart
return MaterialApp.router(
            debugShowCheckedModeBanner: false,
            theme: ThemeData(fontFamily: OptimusFonts.inter),
            locale: flutterLocale,
            supportedLocales: supportedFlutterLocales,
            localizationsDelegates: const [
              GlobalMaterialLocalizations.delegate,
              GlobalWidgetsLocalizations.delegate,
              GlobalCupertinoLocalizations.delegate,
            ],
            routerDelegate: AutoRouterDelegate.declarative(
              _router,
              routes: always(
                [
                  state.when(
                    loading: always(const SplashRoute()),
                    none: always(const NoCredentialsRoute()),
                    provisioning: always(const ProvisioningRoute()),
                    some: (configuration) => ConfiguredAppRoute(
                      configuration: configuration,
                      deviceId: widget.deviceId,
                      application: widget.application,
                      client: widget.client,
                    ),
                  ),
                ],
              ),
            ),
            routeInformationParser: _router.defaultRouteParser(),
          );
```

ConfiguredAppScreen:
```dart
@RoutePage()
class ConfiguredAppScreen extends StatelessWidget {
  const ConfiguredAppScreen({
    Key? key,
    required this.configuration,
    required this.deviceId,
    required this.client,
    required this.application,
  }) : super(key: key);

  final Configuration configuration;
  final String deviceId;
  final String client;
  final String application;

  @override
  Widget build(BuildContext context) => App(
        configuration: configuration,
        deviceId: deviceId,
        application: application,
        client: client,
        child: const AutoRouter(),
      );
}
```

**Comment by Sevastyan** [comment]
Turns out, one must specify a path for the parent route:

```dart
    AutoRoute(
      page: ConfiguredAppRoute.page,
      path: '/configured-app-screen',
      children: [
        AutoRoute(page: SessionRoute.page, path: ''),
        AutoRoute(page: UpdatingRoute.page),
      ],
    )
```

who would have thought... :thinking:

**Comment by Sevastyan** [comment]
Reopening since it feels like a bug in the package.

**Comment by Kerrely** [comment]
Same here, thanks @Sevastyan 🙏  - I wouldn't have thought that adding `path` solve the problem.

**Comment by Milad-Akarie** [comment]
@Sevastyan yup it's a bug! nice catch! fixed in 6.0.1
@Kerrely  just upgrade to the latest build.
